firefoxThemes/popup/popup.js

```diff
@@ -119,29 +99,45 @@ function setTheme(e) {
       const currentMix = chosenThemeName === "mixed" ? mixedStates.INITIAL : mixedStates.NONE;
       let chosenThemeId;
       if (currentMix === mixedStates.NONE) {
+        let themes;
         if (chosenThemeName === "random") {
           chosenThemeId = getRandomTheme(storage.waifuThemes.themes);
+          const chosenRandom = storage.waifuThemes.themes[chosenThemeId]
+          themes = Object.values(storage.waifuThemes.themes)
+            .filter(dokiTheme => (
+              dokiTheme.displayName === chosenRandom.displayName ||
```

Comment:
So this is a bit tricky.

The name used in the drop down is `displayName` however, if there are 2 themes with the same `displayName` then the `name` of the theme is used. There are:
- Hatsune Miku & Nakano Miku
- Tohsaka Rin & Shima Rin

Light & Dark themes of the same character have a name of {{Character Name}} Dark/Light, so to get those the display name is the same.

So to make this more correct, filter by display name && group.

Cause otherwise 2 different Mikus & Rins are going to be hanging out.